File: 12 Conversion/2_binary_decimal.cpp
#include<iostream>
using namespace std;

int main()
{
    int num, mul=1, rem, ans=0;

    cout<<"Enter number: ";
    cin>>num;
    
/*     if (num == 0) {
        cout << 0 << endl;
        return 0;
    } */

    while (num>0)
    {
        rem = num%10;
        num /= 10;
        ans += rem*mul;
        mul *= 2;
    }
    cout << ans << endl;

    return 0;
    
}

File: 012_Conversion/2_binary_decimal.py
n = input("Enter a binary number (only 0s and 1s): ")
num = n

# Check if input contains only 0 and 1
if not all(c in '01' for c in n):
    print("Invalid input! Only binary digits (0 and 1) are allowed.")
    
else:
    n = int(n)
    ans = 0
    m = 1

    while n > 0:
        rem = n%10
        n //= 10
        ans += rem*m
        m *= 2

    print(f"Decimal equivalent {num} : {ans}")
